irdb/publish.py

```diff
@@ -201,58 +281,70 @@ def push_packages_yaml_to_server(login, password):
         Univie u:space username and password
 
     """
-    local_path = PATH_PACKAGES_YAML
-    server_path = "packages.yaml"
-
-    cnopts = pysftp.CnOpts()
-    cnopts.hostkeys = None
-    sftp = pysftp.Connection(host="webspace-access.univie.ac.at",
-                             username=login, password=password, cnopts=cnopts)
-    # with sftp.cd("html/InstPkgSvr/"):
-    with sftp.cd("scopesimu68/html/InstPkgSvr/"):
-        sftp.put(local_path, server_path)
-        print(f"[{str(dt.now())[:19]}]: Pushed to server: packages.yaml")
-
-
-def main(argv):
-    """
-    $ python irdb/publish.py -c -u <PKG_NAME> ... <PKG_NAME_N> -l <USERNAME> -p <PASSWORD>
-    """
-    _pkg_names = []
-    if len(argv) > 1:
-        kwargs = {"compilezip": False, "upload": False, "update_version": True}
-        argv_iter = iter(argv[1:])
-        for arg in argv_iter:
-            if "-" in arg:
-                if "l" in arg:
-                    kwargs["login"] = next(argv_iter)
-                if "p" in arg:
-                    kwargs["password"] = next(argv_iter)
-                if "c" in arg:
-                    kwargs["compilezip"] = "dev" if "dev" in arg else "stable"
-                if "u" in arg:
-                    kwargs["upload"] = True
-                if "d" in arg:
-                    kwargs["update_version"] = False
-                if "h" in arg:
-                    print(HELPSTR)
-                    sys.exit()
-            else:
-                if arg.lower() == "all":
-                    _pkg_names = PKGS.keys()
-                else:
-                    _pkg_names += [arg]
-
-        publish(_pkg_names, **kwargs)
-
-        with open(PATH_PACKAGES_YAML, "w",
-                  encoding="utf8") as f:
-            yaml.dump(PKGS, f)
-
-        if kwargs["upload"]:
-            push_packages_yaml_to_server(login=kwargs["login"],
-                                         password=kwargs["password"])
+    warn(("ANY use of packages.yaml is deprecated. "
+          "No upload will be performed. "
+          "This function will be removed in the next major release."),
+         DeprecationWarning, stacklevel=2)
+
+
+def main():
+    """main CLI script"""
+    parser = argparse.ArgumentParser(prog="publish",
+        description=("Set a new version number, compile (zip) the specified "
+                     "packages and (optionally) push them to the IRDB server. "
+                     "This command must be run from the IRDB root directory."))
+
+    parser.add_argument("pkg_names",
+                        nargs="+",
+                        help="Name(s) of the package(s).")
+    parser.add_argument("-l",
+                        dest="username",
+                        required=True,
+                        help=r"UniVie u:space username - e.g. u\kieranl14.")
+    parser.add_argument("-p",
```

Comment:
Minor nitpicking, the default of `-p` requires one to enter a password even if `-u` is not specified. And `-l` is also required. So just compiling a package now requires something like `python ./irdb/publish.py -c test_package -l Q -p Q`. But that's good enough, since we are the only ones using it. And I believe this was already the case in the old version.